test/unit/spec/common/lang-Spec.js

```diff
@@ -0,0 +1,158 @@
+define(['common/lang'], (Utils) => {
+    'use strict';
+
+    describe('Lang Util functions', () => {
+        it('should have expected functions', () => {
+            ['copy', 'isEqual', 'pRequire'].forEach((fn) => {
+                expect(Utils[fn]).toEqual(jasmine.any(Function));
+            });
+        });
+
+        describe('the copy function', () => {
+            it('should copy a series of objects and such', () => {
+                const testCases = [
+                    'a',
+                    1,
+                    [],
+                    {},
+                    [1, 2, 3],
+                    { a: 1 },
+                    { a: { b: { c: ['d', 'e', 'f'] } } },
+                    undefined,
+                ];
+                testCases.forEach((testCase) => {
+                    expect(Utils.copy(testCase)).toEqual(testCase);
+                });
+            });
+
+            it('altering a copied object should not change the original', () => {
+                const obj = {
+                    foo: {
+                        bar: 'baz',
+                    },
+                };
+
+                const copied = Utils.copy(obj);
+                expect(copied).toEqual(obj);
+                copied.foo.bar = 'wat';
+                expect(copied).not.toEqual(obj);
```

Comment:
could you put in an explicit test:
```js
expect(obj.foo.bar).toEqual('baz');
```

Just to ensure that it's all working as expected

Reply:
Ok, done.